File: spec/javascript/test_nav_parent3.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('NavParent3', 'src/nav_parent2.bin', function(r) {
  assert.equal(r.ofsTags, 8);
  assert.equal(r.numTags, 2);

  assert.equal(r.tags[0].name, 'RAHC');
  assert.equal(r.tags[0].ofs, 0x20);
  assert.equal(r.tags[0].numItems, 3);
  assert.equal(r.tags[0].tagContent.content, 'foo');

  assert.equal(r.tags[1].name, 'RAHC');
  assert.equal(r.tags[1].ofs, 0x23);
  assert.equal(r.tags[1].numItems, 6);
  assert.equal(r.tags[1].tagContent.content, 'barbaz');
});

File: spec/lua/test_nav_parent3.lua
local luaunit = require("luaunit")

require("nav_parent3")

TestNavParent3 = {}

function TestNavParent3:test_nav_parent3()
    local r = NavParent3:from_file("src/nav_parent2.bin")

    luaunit.assertEquals(r.ofs_tags, 8)
    luaunit.assertEquals(r.num_tags, 2)

    luaunit.assertEquals(r.tags[1].name, "RAHC")
    luaunit.assertEquals(r.tags[1].ofs, 0x20)
    luaunit.assertEquals(r.tags[1].num_items, 3)
    luaunit.assertEquals(r.tags[1].tag_content.content, "foo")

    luaunit.assertEquals(r.tags[2].name, "RAHC")
    luaunit.assertEquals(r.tags[2].ofs, 0x23)
    luaunit.assertEquals(r.tags[2].num_items, 6)
    luaunit.assertEquals(r.tags[2].tag_content.content, "barbaz")
end
